subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/DefaultFileWatcherRegistry.java

```diff
@@ -138,6 +139,11 @@ public void handleTerminated() {
         return thread;
     }
 
+    @Override
+    public boolean isWatchingAnyLocations() {
+        return !fileWatcherUpdater.getWatchedFiles().equals(FileHierarchySet.empty());
```

Comment:
That looks weird. Shouldn't we have an `isEmpty()` method on this one?

Reply:
Good point. I added `FileHierarchySet.isEmpty()`.